### PR TITLE
fix(runtimes): Remove unused Pydantic dependencies

### PR DESCRIPTION
**Pull request by jesse-c**
These runtimes don't explicitly use Pydantic. Remove the dependency then to make that clear, and to increase flexibility. A step towards Pydantic v2.

### CODE REVIEW

**Review by mauicv — APPROVED** [review]
LGTM

**Review by sakoush — APPROVED** [review]
LGTM approving assuming comments are sorted before merging:

- I a comment on poetry version to be aligned with what we specify during docker build.

I also see no `pyproject.toml` changes on some runtimes e.g. alibi-explain so I am not sure if they were missed?

File: poetry.lock
Comment (jesse-c) [reply poetry.lock:1]: ✅

**Comment by jesse-c** [comment]
> I also see no pyproject.toml changes for some runtimes e.g. alibi-explain so I am not sure if they were missed?

This is for _unused_ Pydantic dependency declarations. Alibi Explain uses Pydantic [1].

[1] https://github.com/jesse-c/MLServer/blob/02268574ff8cb5d330edfeb8cc10b64896472e22/runtimes/alibi-explain/mlserver_alibi_explain/common.py#L7

**Pull request merged**
